scripts/Muon/GUI/ElementalAnalysis/LoadWidget/load_utils.py

```diff
@@ -117,43 +125,45 @@ def merge_workspaces(run, workspaces):
 def create_merged_workspace(workspace_list):
     if workspace_list:
         # get max number of bins and max X range
-        num_workspaces = len(workspace_list)
         max_num_bins = 0
-        for i in range(0, num_workspaces):
-            ws = mantid.mtd[workspace_list[i]]
-            max_num_bins = max(ws.blocksize(), max_num_bins)
+        for ws_name in workspace_list:
+            if ws_name:
+                ws = mantid.mtd[ws_name]
+                max_num_bins = max(ws.blocksize(), max_num_bins)
 
         # create single ws for the merged data, use original ws as a template
-        merged_ws = mantid.WorkspaceFactory.create(mantid.mtd[workspace_list[0]], NVectors=num_workspaces,
+        template_ws = next(ws for ws in workspace_list if ws is not None)
```

Comment:
:+1: I hadn't considered doing this when you want to use a generator as a filter.